core/changestream/eventsource.go

```diff
@@ -0,0 +1,12 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package changestream
+
+// EventSource describes the ability to subscribe
+// to a subset of events from a change stream.
+type EventSource interface {
```

Comment:
Since this type of interface is quite standard, what do you think about renaming it to `Observable`, as in reactivex? I've also seen this called `Publisher` in the java world, but never `EventSource`. And I'm afraid it might be confusing specially because of it's similarity to the event sourcing pattern/arch.